src/imageio/imageio_qoi.c

```diff
@@ -95,17 +101,20 @@ dt_imageio_retval_t dt_imageio_open_qoi(dt_image_t *img, const char *filename, d
   if(!mipbuf)
   {
     g_free(read_buffer);
-    dt_print(DT_DEBUG_ALWAYS, "[qoi_open] could not alloc full buffer for image: %s\n", img->filename);
+    dt_print(DT_DEBUG_ALWAYS,
+             "[qoi_open] could not alloc full buffer for image: %s\n",
+             img->filename);
     return DT_IMAGEIO_CACHE_FULL;
   }
 
   uint8_t intval;
   float floatval;
+  const size_t npixels = (size_t)desc.width * desc.height;
 
 #ifdef _OPENMP
 #pragma omp parallel for private(intval, floatval)
 #endif
-  for(int i=0; i < desc.width * desc.height * 4; i++)
+  for(int i=0; i < npixels * 4; i++)
```

Comment:
Should be `size_t i`, or we'll truncate here anyway.

Reply:
Indeed. When will I stop doing late-night sleepy PR? :)  Thanks! Fixed.